.github/workflows/build-test-publish.yml

```diff
@@ -102,13 +102,16 @@ jobs:
                         libglu1-mesa-dev libglw1-mesa     \
                         libxm4 build-essential
 
-    - uses: actions/cache@v4
+    - name: Restore cache for AFNI
+      id: cache-afni
+      uses: actions/cache@v4
       with:
         path: /opt/afni
-        key: afni-v1
+        key: afni-v1-${{ matrix.python-version }}-${{ matrix.dependencies }}-${{ matrix.marks }}
         restore-keys: |
-          afni-v1
+          afni-v1-
     - name: Install AFNI
+      if: steps.cache-afni.outputs.cache-hit != 'true'
```

Comment:
We set up this cache to not depend on the specific job, as its contents should not vary. Not saving multiple caches is a feature, not a bug.

Reply:
My memory is vague now, but looking at the commit message maybe GHA was showing some warning about name clashes (e.g. if two processes were trying to write to the same file?)?

Reply:
When there's a cache miss, each job will populate the directory, and the first to finish will successfully save the cache. That's fine and expected. The alternative would be to create a separate job to ensure the cache exists, and insert it into the workflow ahead of these.